### ISSUE TITLE
[Bug]: dropped object still exists in a transaction in RC+ Pessimistic mode 

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#9967]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

[cn.txn]
enable-sacrificing-freshness = true
enable-cn-based-consistency = true
mode = "Pessimistic"
isolation = "RC"

session 1:
mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t1;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)

session 2:
mysql> drop table t1;
Query OK, 0 rows affected (0.03 sec)

mysql> select * from t1;
ERROR 1064 (HY000): SQL parser error: table "t1" does not exist

session 1:
mysql> select * from t1;
+------+
| a    |
+------+
|    1 |
+------+
1 row in set (0.00 sec)

### Expected Behavior

```Markdown
mysql> select * from t1;
ERROR 1064 (HY000): SQL parser error: table "t1" does not exist
```

**Comment by domingozhang** [comment]
If dropping a dropped table in session one, there will be error message after committing:
mysql> drop table t1;
Query OK, 0 rows affected (0.01 sec)

mysql> commit;
ERROR: 
ExpectedEOB

**Comment by volgariver6** [comment]
我先看看这个问题

**Comment by volgariver6** [comment]
@domingozhang 请总结一下ddl的case，然后一起过一下

**Comment by volgariver6** [comment]
正在处理

**Comment by volgariver6** [comment]
pr is committed

**Comment by volgariver6** [comment]
pr需要再完善一下

**Comment by volgariver6** [comment]
fixed by #10045

**Comment by domingozhang** [comment]
fixed
